### PR TITLE
chore: export docfx metadata with build

### PR DESCRIPTION
**Pull request by avafloww**
Adds a build step that runs in parallel with the main build process to export docfx metadata and upload it as a separate workflow artifact.
This will allow us to automatically generate API documentation for the new dev docs site (WIP, will push something up once @goaaats creates the repo for it).

Also changes `fetch-depth` from 0 to 1 to speed up the build process a little bit. The checkout action treats `0` as "fetch all history from every linked repo", which is the opposite of what we want in CI.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -105,3 +105,32 @@ jobs:
         git commit -m "[CI] Update staging for ${DVER} on ${GH_BRANCH}" || true
         
         git push origin main || true
+
+  # docfx metadata generation can often take several minutes, so we run it in parallel to the main build
+  generate-docfx-metadata:
+    name: Generate docfx metadata
+    runs-on: windows-2022
```
Comment (KazWolfe) [inline .github/workflows/main.yml:112]: Do we need to run this task on a windows runner (other than the fact that we already use it)? They're notably slower IME than their Linux alternatives.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ jobs:
         uses: actions/checkout@v3
         with:
           submodules: recursive
-          fetch-depth: 0
+          fetch-depth: 1
```
Comment (KazWolfe) [inline .github/workflows/main.yml:16]: This build's checkout phase took 2m13s vs [36s](https://github.com/goatcorp/Dalamud/actions/runs/5259586413/jobs/9505352763) for a prior PR. Fluke?

Comment (avafloww) [reply .github/workflows/main.yml:16]: Seems so - this only took 26s on my fork.

**Comment by avafloww** [comment]
Closing this and taking a more isolated route to building docs, where the docs repo does most of the build process, instead of relying on metadata generated by the Dalamud build process.

**Pull request closed without merging**